x-pack/legacy/plugins/index_lifecycle_management/public/legacy.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { App } from 'src/core/public';
+
+/* Legacy Imports */
+import { npSetup, npStart } from 'ui/new_platform';
+import chrome from 'ui/chrome';
+import routes from 'ui/routes';
+import { management } from 'ui/management';
+import { fatalError } from 'ui/notify';
+import { createUiStatsReporter } from '../../../../../src/legacy/core_plugins/ui_metric/public';
+
+import { PLUGIN, BASE_PATH } from '../common/constants';
+import { createPlugin } from './np_ready';
+import { addAllExtensions } from './np_ready/extend_index_management';
+
+if (chrome.getInjected('ilmUiEnabled')) {
+  // We have to initialize this outside of the NP lifecycle, otherwise these extensions won't
+  // be available in Index Management unless the user visits ILM first.
+  addAllExtensions();
+
+  /*
+  This method handles the cleanup needed when route is scope is destroyed.  It also prevents Angular
+  from destroying scope when route changes and both old route and new route are this same route.
+  */
+  const manageAngularLifecycle = ($scope: any, $route: any, unmount: () => void) => {
+    const lastRoute = $route.current;
+    const deregister = $scope.$on('$locationChangeSuccess', () => {
+      const currentRoute = $route.current;
+      // if templates are the same we are on the same route
+      if (lastRoute.$$route.template === currentRoute.$$route.template) {
+        // this prevents angular from destroying scope
+        $route.current = lastRoute;
+      }
+    });
+    $scope.$on('$destroy', () => {
+      if (deregister) {
+        deregister();
+      }
+      unmount();
+    });
+  };
+
+  const esSection = management.getSection('elasticsearch');
+  esSection.register('index_lifecycle_policies', {
+    visible: true,
+    display: PLUGIN.TITLE,
+    order: 2,
+    url: `#${BASE_PATH}policies`,
+  });
```

Comment:
FYI once you're app is de-coupled from Angular, you can leverage the new Management plugin in the NP. Can do this in the next PR though.